### PR TITLE
Add tests for classroom

### PR DESCRIPTION
**Pull request by ibugithub**
# 📚 PR: Adding Comprehensive RSpec Unit Tests 🧪

👋 Hi team! This PR enhances our project's reliability by adding robust RSpec unit tests. The tests follow best practices and adhere to the AAA format for improved readability and maintainability. 

### 📖 Classes Tested (all methods):

1. **Book**
2. **Person**
3. **Student**
4. **Teacher**
5. **Classroom**
6. **Rental**
7. **Decorators**

### 🧪 Testing Approach:

- Followed the AAA (Arrange, Act, Assert) pattern for each test.
- Utilized common test doubles to isolate classes and methods.
- Focused on method behavior and edge cases to ensure thorough coverage.

### 🚀 Benefits:

- Improved code quality and reliability through systematic testing.
- Easier debugging and maintenance thanks to detailed test scenarios.
- Enhanced collaboration, as developers can confidently refactor code.
- Reduced risks of introducing new bugs through rigorous testing.

### 🏗️ How to Review:

1. Clone this branch and run the RSpec tests locally.
2. Navigate through each test suite to understand the scope.
3. Check the consistency of naming, formatting, and best practices.
4. Review comments for clarity and thorough coverage.

🙌 Your feedback is crucial! Let's ensure our project's stability and maintainability together. If you have any questions or suggestions, please share them. Let's make our codebase stronger! 💪

📚🔬👨‍🏫👩‍🏫📝🧪🚀

### CODE REVIEW

File: spec/tests_base_decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require '.\base_decorator'
```
Comment (Qoosim) [inline spec/tests_base_decorator_spec.rb]: - [ ] Kindly use correct path in your import.This should be `./base_decorator` NOT `.\base_decorator`. When I test your app with this wrong path, almost your files failed. Kindly check other files and correct the path when needed.

### error
unit-erro

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Some unit tests passed. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/tests_base_decorator_spec.rb (hunk 1 above)
Comment (jicamargo) [reply spec/tests_base_decorator_spec.rb]: ✔️ Changes done. Thank you. 👍🏽

**Review by DuaneDave — APPROVED** [review]
Hi @ibugithub 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ All specs are passing, good job 👍

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
